lib/state/wallet/logic.dart

```diff
@@ -1793,4 +1794,54 @@ class WalletLogic extends WidgetsBindingObserver {
       launchUrl(Uri.parse(uri), mode: LaunchMode.externalApplication);
     } catch (_) {}
   }
+
+  void setWalletActionsLoading(bool loading) {
+    _state.setWalletActionsLoading(loading);
+  }
+
+  Future<void> evaluateWalletActions() async {
+    _state.setWalletActionsLoading(true);
+
+    _state.walletActions = [];
+
+    List<ActionButton> actionsToAdd = [];
+
+    final showVouchers = selectShowVouchers(_state);
+    final showMinter = selectShowMinter(_state);
```

Comment:
I would just make the network request to check if the user has the minter role here and based on the response, add the button.

Remove the network call from the other place it is being done.